### ISSUE TITLE
Sets LANG environment variable to wrong value

### ISSUE THREAD

**Issue by alexreg** [microsoft/vscode#173749]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.75.0
- OS Version: macOS 13.1

Steps to Reproduce:

1. Launch VS Code.
2. Open a new integrated terminal.
3. Enter the command `echo $LANG`
4. Observe that the value of the `LANG` environment variable is wrong; in my case, `en_US.UTF-8` even though it is correctly `en_GB.UTF-8` elsewhere on my system (for example in the Terminal app).

Note, the [same or a very similar issue](https://github.com/microsoft/vscode/issues/13834) was reported back in October 2016.

**Comment by Tyriar** [comment]
@rzhao271 probably a regression from your recent changes in the area?

https://github.com/microsoft/vscode/blob/9e5544bdf5af67ed6ef54d732111330212836027/src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts#L109

**Comment by rzhao271** [comment]
Sorry for the long wait; I finally blocked off some time to look into this issue.

One issue is that we are now passing the app language into the `getLangEnvVariable` function instead of the system locale. Therefore, en-US, the default language VS Code is in, shows up instead of en-GB. However, by setting `LANG` in my .zshrc, I was able to override the integrated terminal's locale in VS Code. I tested using the latest Insiders on macOS Monterey, though, so I might be seeing slightly different behaviour.

@alexreg I'm wondering how you set the `LANG` variable? I was unable to get the built-in Terminal on macOS to print anything when running `echo $LANG` unless I specifically exported it in my .zshrc file.

**Comment by alexreg** [comment]
@rzhao271 Glad to hear; thanks for looking into it.

I don't set the `LANG` variable explicitly anywhere, not in my `.bash_login` or `.bashrc` files nor elsewhere, from what I can tell. May be a difference between Bash and Zsh?

**Comment by rzhao271** [comment]
I'm unable to repro. I switched my shell on macOS to bash, ran `echo $LANG`, and still see an empty string.

**Comment by alexreg** [comment]
Maybe someone else can test too?

**Comment by rzhao271** [comment]
I asked @bpasero to test and he also had an empty `$LANG` string.
I'm wondering whether the workaround of manually setting `$LANG` works for you?

**Comment by alexreg** [comment]
@rzhao271 Yes, I can do the following in my `.brashrc`, though it is of course rather hacky...

```bash
export LANG="$(osascript -e 'user locale of (get system info)').UTF-8"
```

**Comment by weartist** [comment]
@rzhao271 Does a list appear when you type ‘locale’ in the terminal? The first item in the list is the current value of LANG

**Comment by rzhao271** [comment]
@weartist I do get a list, but it just confirms I have an empty value. I get LANG="", LC_COLLATE="C", LC_CTYPE="UTF-8"

**Comment by gusbemacbe** [comment]
I am also having this issue, VSCode shows the wrong `$LANG` variable value on the integrated terminal, while the terminal shows the correct variable value. 

![Image](https://github.com/user-attachments/assets/4a5c4afb-ccde-4c29-8e23-1be6ecaa79e6)
![Image](https://github.com/user-attachments/assets/b0c0b6d2-4316-4789-83ec-ee83e797eda7)
